app/validators/nested_association_validator.rb

```diff
@@ -9,8 +9,11 @@ def initialize(options)
   def validate_each(record, _attribute, value)
     return if !value || value.valid? || value.marked_for_destruction?
     options[:report].each do |alternate, attr|
-      errors = value.errors.messages[alternate]
-      record.errors[attr || alternate].concat(errors) if errors.present?
+      errors = value.errors.where(alternate)
```

Comment:
This block fixes a FrozenError (described [here](https://github.com/rails/rails/issues/37081), and uses a new Error interface, see https://github.com/rails/rails/pull/32313